### ISSUE TITLE
[bug][example] cannot delete phonenumber in field once value is set

### ISSUE THREAD

**Issue by emri99** [cedvdb/phone_form_field#15]
### Problem encountered

In example project only (I don't experimented this in my real project), I can only delete the last character entered and the value is not really deleted.

_Tested on iphone 8 simulator._

It's hard to explain clearly so I made a video ;)

https://user-images.githubusercontent.com/6652177/131548901-df91fdd7-f0e5-4984-bb30-eba27a8b9937.mp4

Sorry I am unable to record pointer events, watch carefully the keyboard area to know which key was pressed, I will edit if I find a better way ;)

**Comment by cedvdb** [comment]
That's probably due to the last update where I introduced several controllers. I hate those, they make the code hard to reason about. I'll take a look

**Comment by emri99** [comment]
I already take a look at that, and I agree with that this is hard to follow too, even if I think that idea is not bad, I didn't achieve fixing this by myself.
You may also check that `onChanged` and `onSaved` are well triggered. In my experiment, tests was failing on theses 2 points.
I experiment that others tests was failing too, but one of my PR fixes some (at least one, I don't remember very well).

**Comment by cedvdb** [comment]
I'm having trouble reproducing this on latest dev. Do you still have the issue ? On which device are you ?


 > You may also check that onChanged and onSaved are well triggered.
yes unit tests are still failing on those 2. I'll fix that in an upcoming PR

**Comment by emri99** [comment]
Still here with last dev branch.
Oh no !... This happens on my real project now :( 

Occurs on iphone 8 simulator (ios 14.5) under MacOS Big Sur 11.3
Will take a look, as it's a really problem that make the field unusable.

Still trouble to reproduce on your side ? What's your test configuration ?

**Comment by cedvdb** [comment]
My test configuration are:

 - latest master
 - android & web

**Comment by cedvdb** [comment]
I have mac os and iphone that I need to setup to see if I can reproduce. I'll set it up tomorrow.

**Comment by emri99** [comment]
I cannot reproduce anymore, tested on latest main, still on iphone 8 simulator.
And this does not occurs anymore on my real project too.

I cannot explain, maybe a local modification... Will investigate & answer here if I find a clue.

Closing, will reopen if needed ;) sorry for the noise.
